whois-api/src/main/java/net/ripe/db/whois/api/elasticsearch/ElasticIndexService.java

```diff
@@ -131,14 +131,20 @@ protected void deleteAll() throws IOException {
         client.deleteByQuery(request, RequestOptions.DEFAULT);
     }
 
+    protected void refreshIndex(){
+        try {
+            client.indices().refresh(new RefreshRequest(whoisAliasIndex), RequestOptions.DEFAULT);
+        } catch (IOException ex){
+            LOGGER.error("Failed to refresh ES index {}: {}", whoisAliasIndex, ex);
+        }
+    }
+
     protected long getWhoisDocCount() throws IOException {
         if (!isElasticRunning()) {
             throw new IllegalStateException("ES is not running");
         }
 
         final CountRequest countRequest = new CountRequest(whoisAliasIndex);
-        final SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
-        searchSourceBuilder.query(QueryBuilders.matchAllQuery());
```

Comment:
I think thats the problem, countRequest needs to setQuery? Can you use setQuery for countRequest and then try out ?

Reply:
CountRequest does not have a setQuery, I checked the documentation and they have source for adding a SearchSourceBUilder which is deprecated
https://www.elastic.co/guide/en/elasticsearch/client/java-rest/current/java-rest-high-count.html
I have already tried this, it is not working. If this was the problem, the refresh will not work. The problem is that the index is not refreshed and is taking previous docs count because the loop that we have takes less than 1 second

Reply:
Sory countRequest takes query builder.  countRequest.query(QueryBuilders.matchAllQuery());

Reply:
ok, let me check. I did not try this

Reply:
Even if it does not work its best to set the queryBuilder in countRequest or check what is the default behaviour.

Reply:
Same issue, I will leave it there.